src/v/ssx/tests/BUILD

```diff
@@ -79,3 +161,79 @@ redpanda_cc_gtest(
         "@seastar",
     ],
 )
+
+redpanda_cc_gtest(
+    name = "work_queue_test",
+    timeout = "short",
+    srcs = [
+        "work_queue_test.cc",
+    ],
+    deps = [
+        "//src/v/base",
+        "//src/v/ssx:work_queue",
+        "//src/v/test_utils:gtest",
+        "@googletest//:gtest",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_gtest(
+    name = "event_test",
+    timeout = "short",
+    srcs = [
+        "event_test.cc",
+    ],
+    deps = [
+        "//src/v/base",
+        "//src/v/ssx:event",
+        "//src/v/test_utils:gtest",
+        "@googletest//:gtest",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_gtest(
+    name = "async_algorithm_test",
+    timeout = "short",
+    srcs = [
+        "async_algorithm_test.cc",
+    ],
+    deps = [
+        "//src/v/base",
+        "//src/v/ssx:async_algorithm",
+        "//src/v/ssx:async_clear",
+        "//src/v/ssx:future_util",
+        "//src/v/test_utils:gtest",
+        "//src/v/utils:move_canary",
+        "@googletest//:gtest",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_bench(
+    name = "thread_worker_bench",
+    timeout = "short",
+    srcs = [
+        "thread_worker_bench.cc",
+    ],
+    deps = [
+        "//src/v/base",
+        "//src/v/ssx:thread_worker",
+        "@seastar",
+        "@seastar//:benchmark",
+    ],
+)
+
+redpanda_cc_bench(
+    name = "sformat_bench",
+    timeout = "short",
+    srcs = [
+        "sformat_bench.cc",
+    ],
+    deps = [
+        "//src/v/base",
```

Comment:
```suggestion
        "//src/v/base",
        "//src/v/random:generators",
```

Reply:
why do we need this? I see it being used anywhere? :thinking:

Reply:
I thought I saw a missing header error in the last failed bazel build. If that's not so, disregard